src/dmd/dtemplate.d

```diff
@@ -2325,8 +2325,10 @@ extern (C++) final class TypeDeduced : Type
  *      tiargs          initial list of template arguments
  *      tthis           if !NULL, the 'this' pointer argument
  *      fargs           arguments to function
+ *      failedIndex     address to store argument index of first type mismatch
  */
-void functionResolve(Match* m, Dsymbol dstart, Loc loc, Scope* sc, Objects* tiargs, Type tthis, Expressions* fargs)
+void functionResolve(Match* m, Dsymbol dstart, Loc loc, Scope* sc, Objects* tiargs,
+    Type tthis, Expressions* fargs, size_t* failedIndex = null)
```

Comment:
Please avoid having default arguments. Explicit is better.

Reply:
It is used elsewhere, e.g. `Parameter.getNth(Parameters* parameters, size_t nth, size_t* pn = null)`. It avoids having to touch existing lines where we want `null`, maybe avoiding merge conflicts.